### PR TITLE
EPMLSTRCMW-258 Feat: Change endpoints in ConfigurationController to RESTful

### PR DESCRIPTION
**Pull request by DruzhininPavel**
**Description:**
Changed endpoints in ConfigurationController to be more RESTful

**Changes:**
ProjectConfigurationController - changed endpoint methods
ProjectConfigurationControllerTest - changed 5 tests
ProjectServiceController - changed addConfiguration method
ProjectServiceControllerTest - changed 3 tests
ProjectConfiguration - added CongugurationAdditionRequest
postman cillection - fixed about new endpoint

### CODE REVIEW

File: services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala [hunk 1]
```diff
@@ -30,25 +28,38 @@ object ProjectConfigurationService {
...
     new ProjectConfigurationService {
 
-      def addConfiguration(projectConfiguration: ProjectConfiguration, userId: UserId): Future[Unit] = {
+      def addConfiguration(
+        projectId: ProjectId,
+        projectConfigurationRequest: ProjectConfigurationAdditionRequest,
+        userId: UserId
+      ): Future[Unit] = {
+
+        val mewProjectConfiguration = ProjectConfiguration(
```
Comment (Nikolai-Isaev) [inline services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala]: "mew" - That's what I call a real cats effect :)

**Review by Nikolai-Isaev — APPROVED** [review]
Looks good but tricky to me

File: services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala (hunk 1 above)
Comment (DruzhininPavel) [reply services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala]: Ooops. I`ll fix it.

**Comment by myazinn** [comment]
I don't think we should change these routes, at least not the way you did it.
`ProjectConfiguration` has its own `id` and is determined by it, not by `projectId`.
If I call `/configurations/some-random-uuid` then I would expect the endpoint to return `ProjectConfiguration` where `id = some-random-uuid`, not `projectId = some-random-uuid`. In our case `configurations/projectId=???` is basically a search query. Moreover, if at some point we decide to allow users to fetch configurations by their id directly, then we will have no way of doing it.
@ilyaonishenko what do you think?

**Comment by DruzhininPavel** [comment]
> I don't think we should change these routes, at least not the way you did it.
> `ProjectConfiguration` has its own `id` and is determined by it, not by `projectId`.
> If I call `/configurations/some-random-uuid` then I would expect the endpoint to return `ProjectConfiguration` where `id = some-random-uuid`, not `projectId = some-random-uuid`. In our case `configurations/projectId=???` is basically a search query. Moreover, if at some point we decide to allow users to fetch configurations by their id directly, then we will have no way of doing it.
> @ilyaonishenko what do you think?

I think we shouldn't think about the endpoints in more detail. Since the entities of the ProjectConfiguration type are nested in the Project, maybe we should put them in the appropriate paths by analogy with something like /project/{{som-project-uuid}}/configurations/{{som-configutaion-uuid}}  this would lead to an understanding of the logic of the endpoints and the correct their construction.

**Review by Nikolai-Isaev — APPROVED** [review]

**Review by gominasian — APPROVED** [review]
LGTM

File: services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
+import cromwell.pipeline.datastorage.dto._
 import cromwell.pipeline.model.wrapper.UserId
 
 import java.nio.file.Path
 import scala.concurrent.{ ExecutionContext, Future }
 
 trait ProjectConfigurationService {
 
-  def addConfiguration(projectConfiguration: ProjectConfiguration, userId: UserId): Future[Unit]
+  def addConfiguration(
```
Comment (myazinn) [inline services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala]: I just realized that it's wrong to change signatures of `Service` methods when we change API scheme. This method should still accept whole `ProjectConfiguration`, now it's up to `Controller` layer to assemble it.

File: controllers/src/test/scala/cromwell/pipeline/controller/ProjectConfigurationControllerTest.scala [hunk 3]
```diff
@@ -21,50 +20,63 @@ class ProjectConfigurationControllerTest extends AsyncWordSpec with Matchers wit
...
 
     "update configuration" should {
       val error = new RuntimeException("Something went wrong")
 
       "return success for update configuration" in {
-        when(configurationService.addConfiguration(configuration, accessToken.userId)).thenReturn(Future.unit)
-        Put("/configurations", configuration) ~> configurationController.route(accessToken) ~> check {
+        when(configurationService.addConfiguration(projectId, configurationAdditionRequest, accessToken.userId))
+          .thenReturn(Future.unit)
+        Put(s"/projects/${projectId.value}/configurations", configurationAdditionRequest) ~> configurationController
```
Comment (myazinn) [inline controllers/src/test/scala/cromwell/pipeline/controller/ProjectConfigurationControllerTest.scala:45]:
```suggestion
        Put(s"/projects/$projectId/configurations", configurationAdditionRequest) ~> configurationController
```
Same for other places

**Review by myazinn — CHANGES_REQUESTED** [review]

File: controllers/src/test/scala/cromwell/pipeline/controller/ProjectConfigurationControllerTest.scala (hunk 3 above)
Comment (DruzhininPavel) [reply controllers/src/test/scala/cromwell/pipeline/controller/ProjectConfigurationControllerTest.scala:45]: If we change this, then we get somthing like this "/project/Project("{PROJECT_UUID}")/configuration"

Comment (myazinn) [reply controllers/src/test/scala/cromwell/pipeline/controller/ProjectConfigurationControllerTest.scala:45]: Oh. Ok, nevermind then

File: services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala (hunk 2 above)
Comment (DruzhininPavel) [reply services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala]: I thout, that controller didn`t create new entities. 
But ok, I can change this.

Comment (myazinn) [reply services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala]: It should not contain businesses logic, but it is allowed to trivially map data from one form to another.
The fact that `projectId` comes from different place than other configuration info is purely caused by our changes of API scheme, no business logic involved. So now it's up to controller layer to make inputs match `Service` layer expectations.

Comment (DruzhininPavel) [reply services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala]: Done

**Review by myazinn — APPROVED** [review]

**Review by Dmitrii-Shevchenko — APPROVED** [review]

**Pull request merged**